<!--
  Standalone launcher for running a default Kobuki
 -->
<launch>
  <arg name="with_arm" default="no" />
  <group ns="group1">
    <arg name="x1" default="-1" />
    <arg name="y1" default="3" />
    <arg name="robot_name1" default="robot_1" />
    <arg name="urdf_file" default="$(find xacro)/xacro '$(find hola_tortuga)/urdf/kobuki_standalone.urdf.xacro' x:=$(arg x1) y:=$(arg y1) tf_prefix:=$(arg robot_name1)_tf with_arm:=$(arg with_arm)" />
    <param name="robot_description" command="$(arg urdf_file)" />

    <node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
    <node pkg="nodelet" type="nodelet" name="mobile_base" args="load kobuki_softnode/SoftKobukiNodelet mobile_base_nodelet_manager">
      <rosparam file="$(find kobuki_softnode)/param/base.yaml" command="load"/>

      <!-- Remap needed to use the custom tf_prefix -->
      <param name="odom_frame" type="string" value="$(arg robot_name1)_tf/odom"/>
      <param name="base_frame" type="string" value="$(arg robot_name1)_tf/base_footprint"/>


      <remap from="mobile_base/odom" to="$(arg robot_name1)/odom"/>
      <!-- <remap from="mobile_base/joint_states" to="joint_states"/> -->
    </node>

    <!-- <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >
      <rosparam command="load" file="$(find kobuki_softnode)/param/diagnostics.yaml" />
      <param name="tf_prefix" type="string" value="$(arg robot_name)_tf"/>
    </node> -->

    <!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
      <param name="tf_prefix" value="$(arg robot_name1)_tf"/>
      <param name="robot_description"  command="$(arg urdf_file)"/>
      <rosparam param="source_list">["joint_states_interpolated"]</rosparam>
      <param name="source_list" value="[joint_states]" />
    </node> -->

    <node pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" name="joint_state_publisher_gui" output="screen">
      <param name="publish_frequency" type="double" value="30.0" />
    </node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="30.0" />
    </node>

    <!-- <include file="$(find kobuki_keyop)/launch/keyop.launch" /> -->

  </group>

  <group ns="group2">
    <arg name="x2" default="3" />
    <arg name="y2" default="2" />
    <arg name="robot_name2" default="robot_2" />
    <arg name="urdf_file" default="$(find xacro)/xacro '$(find hola_tortuga)/urdf/kobuki_standalone.urdf.xacro' x:=$(arg x2) y:=$(arg y2) tf_prefix:=$(arg robot_name2)_tf with_arm:=$(arg with_arm)" />
    
    <param name="robot_description" command="$(arg urdf_file)" />

    <node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
    <node pkg="nodelet" type="nodelet" name="mobile_base" args="load kobuki_softnode/SoftKobukiNodelet mobile_base_nodelet_manager">
      <rosparam file="$(find kobuki_softnode)/param/base.yaml" command="load"/>
      <param name="odom_frame" type="string" value="$(arg robot_name2)_tf/odom"/>
      <param name="base_frame" type="string" value="$(arg robot_name2)_tf/base_footprint"/>


      <remap from="mobile_base/odom" to="$(arg robot_name2)/odom"/>
      <!-- <remap from="mobile_base/joint_states" to="joint_states"/> -->
    </node>

    <!-- <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >
      <rosparam command="load" file="$(find kobuki_softnode)/param/diagnostics.yaml" />
      <param name="tf_prefix" type="string" value="$(arg robot_name)_tf"/>
    </node> -->

    <!-- <node name="joint_state_publisher" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" >
      <param name="tf_prefix" type="string" value="$(arg robot_name2)_tf"/>
    </node> -->

    <node pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" name="joint_state_publisher_gui" output="screen">
      <param name="publish_frequency" type="double" value="30.0" />
    </node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="30.0" />
      <!-- <param name="tf_prefix" type="string" value="$(arg robot_name2)_tf"/> -->
    </node>

    <!-- <include file="$(find kobuki_keyop)/launch/keyop.launch" />   -->

  </group>


	<arg name="rvizconfig" default="$(find hola_tortuga)/rviz/full_custom_launch.rviz" />

	<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

</launch>
